File: Python/oop/bike/bike.py
# bike.py

class bike:
    def __init__(self,price,max_speed):
        self.price = price
        self.max_speed = max_speed
        self.miles = 0
        self.p = lambda x : print(x)
    def displayInfo(self):
        s = 'price: {} max speed: {} total miles: {}'.format(self.price, self.max_speed, self.miles)
        self.p(s)
        return self
    def ride(self):
        self.miles += 10
        self.p('Riding')
        return self 
    def reverse(self):
        self.miles -= 5 
        self.p('Reversing')
        return self 

def run():
    inst1 = bike(500,200)
    inst1.ride().ride().ride().reverse().displayInfo()
    inst2 = bike(100,100)
    inst2.ride().ride().reverse().reverse().displayInfo()
    inst3 = bike(10,10)
    inst3.reverse().reverse().reverse().displayInfo()


run()

File: MEAN/angular/typescript/oop-bike/bike.ts
class Bike{
    public miles : number; 
    constructor(public price : number, public max_speed : String){
        this.miles = 0
    }

    displayInfo(){
        console.log(`the bikes price ${this.price} 
        the max speed ${this.max_speed} 
        the total miles ${this.miles}`)
        return this 
    }

    ride(){
        console.log('Riding....')
        this.miles += 10
        return this 
    }

    reverse(){
        console.log(' Reversing.......')
        this.miles -= 5
        return this 
    }

}

let bike1 = new Bike(300, '225mph')
let bike2 = new Bike(250, '250mph')
let bike3 = new Bike(350, '450mph')

bike1.ride().ride().ride().displayInfo()


